cpp/src/arrow/python/python_test.cc

```diff
@@ -194,7 +194,7 @@ TEST(PyBuffer, InvalidInputObject) {
 // ("unresolved external symbol arrow_ARRAY_API referenced").
 #ifndef _WIN32
 TEST(PyBuffer, NumpyArray) {
-  const npy_intp dims[1] = {10};
+  npy_intp dims[1] = {10};
```

Comment:
What are these changes for?

Reply:
I do not recall if it was in CI or a compiler on my side, but build was triggering error w.r.t. to `const npy_intp *` in  `PyArray_SimpleNew`. I had not seen this issue before and the `const` attribute is present in the latest NumPy API but it was not there at some point. I revert this change, but note that this TEST uses `const npy_intp` and the following below does not.